src/ess/loki/general.py

```diff
@@ -6,50 +6,44 @@
 
 import sciline
 import scipp as sc
-from ess.reduce import nexus
+from ess.reduce.nexus.generic_workflow import GenericNeXusWorkflow
 
 from ess.sans import providers as sans_providers
+from ess.sans.io import read_xml_detector_masking
 
-from ..sans.common import gravity_vector
 from ..sans.types import (
-    BeamCenter,
-    CalibratedDetector,
     CorrectForGravity,
+    DetectorBankSizes,
     DetectorData,
-    DetectorEventData,
     DetectorPixelShape,
     DimsToKeep,
     Incident,
     LabFrameTransform,
-    MonitorEventData,
+    MonitorData,
     MonitorType,
     NeXusDetector,
-    NeXusMonitor,
     NeXusMonitorName,
     NonBackgroundWavelengthRange,
     PixelShapePath,
-    RawDetector,
-    RawMonitor,
-    RawMonitorData,
-    RawSample,
-    RawSource,
     RunType,
-    SamplePosition,
     ScatteringRunType,
-    SourcePosition,
     TofData,
     TofMonitor,
     TransformationPath,
     Transmission,
     WavelengthBands,
     WavelengthMask,
 )
-from .io import dummy_load_sample
+
+DETECTOR_BANK_RESHAPING = {
```

Comment:
```suggestion
DETECTOR_BANK_SIZES = {
```
(this is the [name used](https://github.com/scipp/essdiffraction/blob/main/src/ess/dream/io/nexus.py#L20) in `essdiffraction`)